src/main/java/org/tron/core/db/TransactionTrace.java

```diff
@@ -175,13 +166,14 @@ public void pay() throws BalanceInsufficientException {
       case TRX_CONTRACT_CALL_TYPE:
         TriggerSmartContract callContract = ContractCapsule
             .getTriggerContractFromTransaction(trx.getInstance());
-        callerAccount = callContract.getOwnerAddress().toByteArray();
-
-        ContractCapsule contract =
+        ContractCapsule contractCapsule =
             dbManager.getContractStore().get(callContract.getContractAddress().toByteArray());
-        originAccount = contract.getInstance().getOriginAddress().toByteArray();
-        percent = Math.max(100 - contract.getConsumeUserResourcePercent(), 0);
+
+        callerAccount = callContract.getOwnerAddress().toByteArray();
+        originAccount = contractCapsule.getOriginAddress();
+        percent = Math.max(100 - contractCapsule.getConsumeUserResourcePercent(), 0);
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Assign this magic number 100 to a well-named constant, and use the constant instead. ([squid:S109](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS109))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1670&issues=AWbSdHdCSgN4uCclEWob&open=AWbSdHdCSgN4uCclEWob)

[ISSUE_KEY]: AWbSdHdCSgN4uCclEWob